x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -229,25 +229,10 @@ public static boolean isNullOrNumeric(DataType t) {
         return t.isNumeric() || isNull(t);
     }
 
-    public static boolean isSigned(DataType t) {
-        return t.isNumeric() && t.equals(UNSIGNED_LONG) == false;
-    }
-
     public static boolean isDateTime(DataType type) {
         return type == DATETIME;
     }
 
-    public static boolean areCompatible(DataType left, DataType right) {
```

Comment:
I think we'll want to keep this method here and remove the equivalent from `EsqlDataTypes.java`, which might be at some point [removed entirely](https://github.com/elastic/elasticsearch/pull/109713#discussion_r1643455820) (similar to how `isString()` is kept here and has been removed from there). But this can be done at some subsequent refactor.
LGTM otherwise.

Reply:
The implementation of the two is slightly different:

https://github.com/elastic/elasticsearch/blob/39aa832400b526f41e0019029e42d04efbfcdcb5/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypes.java#L85-L93

I did not know what are the consequences of having the extra ` || (isDateTime(left) && isDateTime(right));` condition.

Reply:
> I did not know what are the consequences of having the extra || (isDateTime(left) && isDateTime(right)); condition.

Right now, none. Since ESQL (unlike SQL, where that check was likely added) only has one datetime type (`DATETIME`), the check would be mostly superfluous, since for it to match, both types would have to be the same, so caught by the first equality check.

But once we'll have more than one type (like the upcoming [nanos support](https://github.com/elastic/elasticsearch/issues/109352), we'll likely need to add this check back. CC: @not-napoleon.